src/pages/stop/stop.component.ts

```diff
@@ -112,7 +116,38 @@ export class StopComponent {
     });
   }
 
+  getAlerts(): void {
+    this.alerts = [];
+    this.alertService
+    .getAlerts().then(alerts => {
+      if (!alerts) {
+        return;
+      }
+      for (let alert of alerts) {
+        // display alerts that don't apply to any route
+        if (alert.Routes.length === 0) {
+          if (!_.includes(this.alerts, alert)) {
+            this.alerts.push(alert);
+          }
+        } else {
+          // display alerts that apply to routes that service the stop
+          this.stopDepartureSvc.getStopDeparture(this.stopId)
+            .then(directions => {
+              for (let route of directions[0].RouteDirections) {
```

Comment:
Think I did this part wrong

Reply:
I mean, probably not wrong. Because it does what I want it to. But idk if getting the stop departure is the most efficient/smartest way to do this.

Reply:
I see what you mean, because of course, the app also fetches those stop departures for purposes of actually displaying departures too.